mlflow/pyfunc/__init__.py

```diff
@@ -425,6 +441,11 @@ def predict(self, data: PyFuncInput) -> PyFuncOutput:
                 if _MLFLOW_TESTING.get():
                     raise
 
+        # Models saved prior to MLflow 2.5.0 do not support `params` in the pyfunc `predict()`
+        # function definition, nor do they support `**kwargs`. Accordingly, we only pass `params`
+        # to the `predict()` method if it defines the `params` argument
+        if inspect.signature(self._predict_fn).parameters.get("params"):
```

Comment:
Should we add a warning here similar to https://github.com/mlflow/mlflow/blob/eb621e58e81707254341837a9ba1240b3a381ce1/mlflow/pyfunc/__init__.py#L428-L432

Reply:
I think that's a good idea; otherwise users will wonder why things aren't changing if we silently ignore their inputs :)